xradar/src/main/java/com/orzangleli/radar/XRadarView.java

```diff
@@ -251,40 +262,60 @@ public XRadarView(Context context, @Nullable AttributeSet attrs, int defStyleAtt
         percents = new double[count];
         values = new CharSequence[count];
 
+        addDrawTask(this);
+        setLayoutRefreshedListener(new RefreshListener());
+        setTouchEventListener(this);
     }
 
     @Override
-    protected void onSizeChanged(int w, int h, int oldw, int oldh) {
-        radius = Math.min(h, w) / 2 * radarPercent;
-        MAX_TEXT_WIDTH = (int) (Math.min(h, w) / 2 * (1 - radarPercent));
-        //中心坐标
-        centerX = w / 2;
-        centerY = h / 2;
-        postInvalidate();
-        super.onSizeChanged(w, h, oldw, oldh);
-    }
-
-    @Override
-    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
-        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
-        int wSpecSize = MeasureSpec.getSize(widthMeasureSpec);
-        int hSpecSize = MeasureSpec.getSize(heightMeasureSpec);
-
-        setMeasuredDimension(Math.min(wSpecSize, hSpecSize), Math.min(wSpecSize, hSpecSize));
-    }
+    public boolean onTouchEvent(Component component, TouchEvent touchEvent) {
+        HiLog.info(LABEL_LOG, "%{public}s", "onTouchEvent called");
+        switch (touchEvent.getAction()) {
+            case TouchEvent.PRIMARY_POINT_DOWN:
+                int x = (int) touchEvent.getPointerPosition(touchEvent.getIndex()).getX();
+                int y = (int) touchEvent.getPointerPosition(touchEvent.getIndex()).getY();
+                for (int i = 0; i < titleRects.size(); i++) {
+                    Rect rect = titleRects.get(i);
+                    if (rect != null && rect.contains(x, y, x, y)) {
+                        if (onTitleClickListener != null) {
+                            onTitleClickListener.onTitleClick(XRadarView.this, i, x, y, rect);
+                            return true;
+                        }
+                    }
+                }
+                break;
+        }
+        return false;
+    }
+
+    /**
+     * RefreshListener updates when view is relayout
+     */
+    class RefreshListener implements LayoutRefreshedListener {
+        @Override
+        public void onRefreshed(Component component) {
+            int w = component.getWidth();
+            int h = component.getHeight();
+            radius = Math.min(h, w) / 2 * radarPercent;
+            MAX_TEXT_WIDTH = (int) (Math.min(h, w) / 2 * (1 - radarPercent));
```

Comment:
Here there is a numeric issue - integer division in the floating-point context. Also, similar issues on several places, pls fix all.

Reply:
solved numeric warning - integer division in the floating-point context.